### PR TITLE
Moved `file.open` command to `@theia/core`. #1609

### PR DESCRIPTION
**Pull request by marechal-p**
Signed-off-by: Paul Maréchal <paul.marechal@ericsson.com>

Refactoring based on #1609 .

Shouldn't we make the `opener-service.open` function a `OpenerService.open` method ?

### CODE REVIEW

**Comment by kittaakos** [comment]
I have restarted the builds and doing the review now.

File: packages/core/src/common/uri-command.ts [hunk 1]
```diff
@@ -0,0 +1,140 @@
+/*
```
Comment (kittaakos) [inline packages/core/src/common/uri-command.ts]: Could you please rename the `uri-command` module to `uri-command-handler`?

**Comment by kittaakos** [comment]
> Shouldn't we make the opener-service.open function a OpenerService.open method ?

Sorry, I have missed that. If I am not replying such questions, just mention me.

If you think it will better, then be my guest. Change it and merge the `open` function into the `open` method of the `OpenerService`.

Why do we have the `open` function instead of a method; I assume, legacy.

**Comment by marechal-p** [comment]
Playing around with this `open` function, it is hard to not leave it on the side like it is right now.

At least I didn't find an elegant way of doing it, so if you don't have any suggestion on how to move it then it would be the last step before merging.

If its all good as it is, then I'm cool with it too !

**Comment by kittaakos** [comment]
Sorry, I did not fully get this.
 - Did you decide not to do this? I am absolutely OK leaving it as is. The task and the PR were about the open command and its handler anyway.
 - Or shall we figure out something out together? If so, what can I help?

**Comment by marechal-p** [comment]
I just didn't find a way to move the `open` function to be a `OpenService.open` method:

- `OpenService` is an interface, and a symbol too. (Can't convert it directly to a class)
- If I move `open` inside the interface, it is without implementation. I could put the implementation inside the `DefaultOpenerService` that seems to currently be the only `OpenerService` implementation, but I don't know if people will be likely to create their own, and if they will want to use the open function implementation on their new `OpenerService`, in which case having `open` on the side makes it sharable between multiple opener services. But it will feel like it should belong to an `OpenerService` because it is meant to be run on opener services.

I was just stuck on this small detail, I was hoping you could just cut the pie for me on this question :p

**Comment by kittaakos** [comment]
Indeed, we cannot do much with that. One could move the `open` function under the `OpenerService` namespace, but that would not change anything. Instead of importing and calling `open`, it would be `OpenerService.open`. We're done here.

**Pull request merged**